### PR TITLE
Add staged.dependencies

### LINKED ISSUE #33: add staged depenency
with `synthetic.cdisc.data` as a downstream dependency

### PR DESCRIPTION
**Pull request by nikolas-burkoff**
Closes #33 

Replaces master with main in gh action workflows as main is now the default branch

Replaced the current build-check-install with one which uses staged.dependencies (tweaked to install curl) - also **removed rocker 3.6.3 from the build-check-install** as it fails on the docker file requirements check
image
and if you remove that check then it still fails e.g. as the version of git on the rocker image is too old for `git branch --show-current`
image

Suggest putting an issue in the appropriate place if we need the gh action to work with rocker 3.6.3 image

### CODE REVIEW

File: .github/workflows/build-check-install.yaml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  build-install-check:
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ matrix.config.image }}:${{ matrix.config.tag }}
+    name: ${{ matrix.config.image }}, version ${{ matrix.config.tag }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {image: rocker/tidyverse, tag: '4.1.0'}
```
Comment (dinakar29) [inline .github/workflows/build-check-install.yaml:21]: There might be better CI performance with `{image: ghcr.io/insightsengineering/tidyverse_4.1.0_bioc_3.13, tag: latest}`
Please give that a shot and do a benchmark test.

Comment (nikolas-burkoff) [reply .github/workflows/build-check-install.yaml:21]: I couldn't get it to work I'm afraid https://github.com/Roche/synthetic.cdisc.data/runs/3191977819?check_suite_focus=true - any thoughts?

Comment (dinakar29) [reply .github/workflows/build-check-install.yaml:21]: I think I fixed it - visibility settings needed to be updated. Try now?

Comment (nikolas-burkoff) [reply .github/workflows/build-check-install.yaml]: Yeah that works - thanks

**Comment by dinakar29** [comment]
@nikolas-burkoff Let's leave out the R CMD check PR comment posting step in the workflow. It doesn't seem to perform correctly and is likely to fail in future runs if there are unescaped characters that Node.js cannot parse.

**Comment by nikolas-burkoff** [comment]
> @nikolas-burkoff Let's leave out the R CMD check PR comment posting step in the workflow. It doesn't seem to perform correctly and is likely to fail in future runs if there are unescaped characters that Node.js cannot parse.

👍 - removed

**Review by dinakar29 — APPROVED** [review]
LGTM

**Comment by nikolas-burkoff** [comment]
@waddella  - will merge first thing 3rg Aug unless I hear from you

File: .github/workflows/build-check-install.yaml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+
+      - name: Install R package dependencies
+        run: |
+          setwd("${{ github.event.repository.name }}")
+          options(repos = c(CRAN = "https://cloud.r-project.org/"))
+          ncores <- parallel::detectCores(all.tests = FALSE, logical = TRUE)
+          cat(paste("\n\nnumber of cores detected:", ncores, "\n\n"))
+          if (file.exists("renv.lock")) {
+            renv::restore()
```
Comment (waddella) [inline .github/workflows/build-check-install.yaml:114]: please add

```
if (!require("renv")) install.packages("renv",  upgrade = "never", Ncpus = ncores)
```

**Review by waddella — APPROVED** [review]
please debug and add the line in my review

**Pull request merged**